File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
# 1 - 2 - 3 - 5

def solution(n):
    if n == 1: return 1
    elif n == 2: return 2

    answer = [0] * (n + 1)
    answer[1] = 1
    answer[2] = 2
    index = 3
    
    while not answer[n]:
        answer[index] = (answer[index - 2] + answer[index - 1]) % 1234567
        index += 1
    
    return answer[n]

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    const dp = Array(2001).fill(0);
    dp[1] = 1;
    dp[2] = 2;
    
    for (let i = 3; i <= n; i++) {
        dp[i] = (dp[i - 2] % 1234567) + (dp[i - 1] % 1234567);
    }    
    
    return dp[n] % 1234567;
}